rbac/management/utils.py

```diff
@@ -115,7 +115,7 @@ def get_principal(
         else:
             # Avoid possible race condition if the user was created while checking BOP
             principal, _ = Principal.objects.get_or_create(
-                username=username, tenant=tenant
+                username=username, tenant=tenant, user_id=request.user.user_id
```

Comment:
Will we need to ensure we backfill data first? Wondering if this would cause a false-negative lookup and end up creating duplicates in the case where:

- We have a principal with `username="Foo", user_id=null` because we've merged this but not backfilled `user_id` values
- This lookup gets called and now `reqeust.user.user_id` is populated, so the lookup might be on `username="Foo", user_id=123` which would create a new record?